src/haz3lcore/lang/term/Grammar.re

```diff
@@ -417,3 +417,400 @@ and map_type_hole_annotation:
     | MultiHole(l) => MultiHole(List.map(x => map_any_annotation(f, x), l))
     };
   };
+
+module type DefaultAnnotation = {
+  type t;
+  let default_value: unit => t;
+};
```

Comment:
You could have 2 types here `t` for the type of the annotation and another type `s` that could be used in the factory methods below and require a `default_value : option(s) -> t` as part of the builder. I haven't bothered but for optional annotations it allows the people calling the builder to use a different type like not provide `Some` constructors.